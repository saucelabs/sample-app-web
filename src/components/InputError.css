.form_group {
  margin-bottom: 15px;
  position: relative;
}
.input_error {
  font-family: "Roboto", Arial, Helvetica, sans-serif;
  font-size: 18px;
  width: 100%;
  border: 0px;
  border-bottom: 2px solid #ededef;
  outline: none;
  padding: 10px 0;
}

.input_error.error {
  border-bottom-color: #e2231a;
}

.error_icon {
  color: #e2231a;
  font-size: 18px;
  position: absolute;
  right: 0;
  top: 12px;
}

/* Chrome/Opera/Safari */
.form_input::-webkit-input-placeholder,
    /* Firefox 19+ */
.form_input::-moz-placeholder,
    /* IE 10+ */
.form_input:-ms-input-placeholder,
.form_input::placeholder {
  color: #c1c7c6;
}

/* Chrome/Opera/Safari */
.form_input.error::-webkit-input-placeholder,
    /* Firefox 19+ */
.form_input.error::-moz-placeholder,
    /* IE 10+ */
.form_input.error:-ms-input-placeholder,
.form_input.error::placeholder {
  color: #e2231a !important;
}
