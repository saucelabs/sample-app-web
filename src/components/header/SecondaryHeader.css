.header_secondary_container {
  background: #474c55;
  width: 100%;
  height: 80px;
  display: flex;
  flex-direction: row;
  align-items: center;
  padding: 0 30px;
  box-sizing: border-box;
}

.header_secondary_container .title {
  color: #fff;
  font-size: 24px;
  text-transform: uppercase;
}

.header_secondary_container .peek {
  background: url("../../assets/img/headerBot2x.png") no-repeat center center;
  background: url("../../assets/svg/headerBot3x.svg") no-repeat center center,
    linear-gradient(transparent, transparent);
  display: block;
  height: 91px;
  width: 90px;
  margin-top: 13px;
  margin-left: 25px;
}

.right_component {
  margin-left: auto;
}

@media only screen and (max-width: 640px) {
  .header_secondary_container {
    height: 60px;
    padding: 0 15px;
  }

  .header_secondary_container .title {
    font-size: 20px;
  }

  .header_secondary_container .peek {
    display: none;
  }
}
