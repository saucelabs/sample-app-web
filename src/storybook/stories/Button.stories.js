import React, { useState } from "react";

import Button, { BUTTON_SIZES, BUTTON_TYPES } from "../../components/Button";

export default {
  title: "SwagLabs/Button",
  component: Button,
};

const Template = (args) => {
  const [value, setValue] = useState(0);

  return (
    <>
      <Button {...args} onClick={() => setValue(value + 1)} />
      <p>
        You clicked <strong>{value}</strong> times.
      </p>
    </>
  );
};

export const Primary = Template.bind({});
Primary.args = {
  label: "Primary",
  size: BUTTON_SIZES.MEDIUM,
  type: BUTTON_TYPES.PRIMARY,
};

export const Secondary = Template.bind({});
Secondary.args = {
  label: "Secondary",
  size: BUTTON_SIZES.MEDIUM,
  type: BUTTON_TYPES.SECONDARY,
};

export const Action = Template.bind({});
Action.args = {
  label: "Some Action",
  size: BUTTON_SIZES.MEDIUM,
  type: BUTTON_TYPES.ACTION,
};

export const Back = Template.bind({});
Back.args = {
  label: "Go back",
  size: BUTTON_SIZES.MEDIUM,
  type: BUTTON_TYPES.BACK,
};

export const Small = Template.bind({});
Small.args = {
  label: "Small",
  size: BUTTON_SIZES.SMALL,
};

export const Medium = Template.bind({});
Medium.args = {
  label: "Medium button",
  size: BUTTON_SIZES.MEDIUM,
};

export const Large = Template.bind({});
Large.args = {
  label: "Large button",
  size: BUTTON_SIZES.LARGE,
};
