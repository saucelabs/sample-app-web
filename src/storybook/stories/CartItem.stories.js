import React from "react";
import StoryRouter from "storybook-react-router";
import CartItem from "../../components/CartItem";

export default {
  title: "SwagLabs/Swag Cart Item",
  component: CartItem,
  decorators: [StoryRouter()],
  parameters: {
    layout: "centered",
  },
};

const Template = (args) => (
  <>
    <CartItem {...args} />
    <p>Resize the preview to see the mobile view.</p>
  </>
);

export const SwagCartItem = Template.bind({});
SwagCartItem.args = {
  history: {
    push: () => {},
  },
  item: {
    id: 1,
    name: "Sauce Labs Bolt T-Shirt",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
  },
  showButton: true,
};

export const SwagCartItemNoButton = Template.bind({});
SwagCartItemNoButton.args = {
  history: {
    push: () => {},
  },
  item: {
    id: 1,
    name: "Sauce Labs Bolt T-Shirt",
    desc: `Get your testing superhero on with the Sauce Labs bolt T-shirt. From American Apparel, 100% ringspun combed cotton, heather gray with red bolt.`,
    price: 15.99,
  },
  showButton: false,
};
