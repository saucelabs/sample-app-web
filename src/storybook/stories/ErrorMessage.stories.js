import React, { useState } from "react";
import ErrorMessage from "../../components/ErrorMessage";

export default {
  title: "SwagLabs/Form/ErrorMessage",
  component: ErrorMessage,
};

const Template = (args) => {
  const [value, setValue] = useState(0);
  return (
    <>
      <ErrorMessage {...args} onClick={() => setValue(value + 1)} />
      <p>
        You clicked <strong>{value}</strong> times.
      </p>
    </>
  );
};

export const Error = Template.bind({});
Error.args = {
  isError: true,
  errorMessage: "This is an error message",
  onClick: () => console.log("clicked"),
};
