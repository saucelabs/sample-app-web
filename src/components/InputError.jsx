import React from "react";
import PropTypes from "prop-types";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimesCircle } from "@fortawesome/free-solid-svg-icons";
import "./InputError.css";

export const INPUT_TYPES = {
  TEXT: "text",
  PASSWORD: "password",
};
const InputError = ({
  isError,
  onChange,
  placeholder,
  testId,
  type,
  value,
  ...props
}) => {
  return (
    <div className="form_group">
      <input
        // `form_input` has no style function
        // but is there for backwards compatibility
        className={`input_error form_input${isError ? " error" : ""}`}
        placeholder={placeholder}
        onChange={onChange}
        type={type}
        value={value}
        {...(testId
          ? {
              "data-test": testId,
              id: testId,
              name: testId,
            }
          : {})}
        {...props}
      />
      {isError && (
        <FontAwesomeIcon icon={faTimesCircle} className="error_icon" />
      )}
    </div>
  );
};

InputError.propTypes = {
  /**
   * If this is an isError field yes or no
   */
  isError: PropTypes.bool.isRequired,
  /**
   * The on change handler
   */
  onChange: PropTypes.func.isRequired,
  /**
   * The placeholder of the input
   */
  placeholder: PropTypes.string,
  /**
   * The test id
   */
  testId: PropTypes.string,
  /**
   * What type of field is it
   */
  type: PropTypes.oneOf(["text", "password"]),
  /**
   * The value of the input
   */
  value: PropTypes.string,
};

InputError.defaultProps = {
  placeholder: "",
  testId: undefined,
  type: INPUT_TYPES.TEXT,
  value: "",
};

export default InputError;
