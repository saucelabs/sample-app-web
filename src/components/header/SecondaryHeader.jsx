import React from "react";
import PropTypes from "prop-types";
import "./SecondaryHeader.css";

const SecondaryHeader = ({
  headerBot,
  leftComponent,
  rightComponent,
  title,
}) => {
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const HeaderBot = () => <div className="peek" />;
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const LeftComponent = ({ leftComponent }) => (
    <div className="left_component">{React.cloneElement(leftComponent)}</div>
  );
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const RightComponent = ({ rightComponent }) => (
    <div className="right_component">{React.cloneElement(rightComponent)}</div>
  );
  /**
   * @TODO:
   * This can't be tested yet because enzyme currently doesn't support ReactJS17,
   * see https://github.com/enzymejs/enzyme/issues/2429.
   * This means we can't fully mount the component and test all rendered components
   * and functions
   */
  /* istanbul ignore next */
  const Title = ({ title }) => <span className="title">{title}</span>;

  return (
    <div className="header_secondary_container">
      {leftComponent && <LeftComponent leftComponent={leftComponent} />}
      {title && <Title title={title} />}
      {headerBot && <HeaderBot />}
      {rightComponent && <RightComponent rightComponent={rightComponent} />}
    </div>
  );
};
SecondaryHeader.propTypes = {
  /**
   * Show the header bot
   */
  headerBot: PropTypes.bool,
  /**
   * A react component for the left
   */
  leftComponent: PropTypes.element,
  /**
   * A react component for the right
   */
  rightComponent: PropTypes.element,
  /**
   * A title
   */
  title: PropTypes.string,
};
SecondaryHeader.defaultProps = {
  headerBot: undefined,
  leftComponent: undefined,
  rightComponent: undefined,
  title: undefined,
};

export default SecondaryHeader;
