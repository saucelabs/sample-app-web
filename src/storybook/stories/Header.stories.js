import React from "react";
import SecondaryHeader from "../../components/header/SecondaryHeader";
import Button, { BUTTON_TYPES } from "../../components/Button";

export default {
  title: "SwagLabs/Headers",
  component: SecondaryHeader,
};

const Template = (args) => (
  <>
    <SecondaryHeader {...args} />
  </>
);

export const SecondaryEmpty = Template.bind({});
SecondaryEmpty.args = {};

export const SecondaryLeftComponent = Template.bind({});
SecondaryLeftComponent.args = {
  leftComponent: (
    <Button
      label="Back to Products"
      onClick={() => {}}
      type={BUTTON_TYPES.BACK}
    />
  ),
};

export const SecondaryTitle = Template.bind({});
SecondaryTitle.args = {
  title: "Title only",
};

export const SecondaryTitleBotRight = Template.bind({});
SecondaryTitleBotRight.args = {
  headerBot: true,
  rightComponent: (
    <Button
      label="Go to something"
      onClick={() => {}}
      type={BUTTON_TYPES.ACTION}
    />
  ),
  title: "Title and Bot",
};
