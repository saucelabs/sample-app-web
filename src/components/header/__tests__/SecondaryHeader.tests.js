import React from "react";
import { shallow } from "enzyme";
import SecondaryHeader from "../SecondaryHeader";
import { Input } from "../../../storybook/stories/InputError.stories";

let props;

describe("SecondaryHeader", () => {
  it("should render without any props", () => {
    const component = shallow(<SecondaryHeader />);

    expect(component).toMatchSnapshot();
  });

  it("should render all props", () => {
    const component = shallow(
      <SecondaryHeader
        headerBot
        leftComponent={<Input type="checkbox" />}
        rightComponent={<Input type="radio" />}
        title="sample title"
      />
    );

    expect(component).toMatchSnapshot();
  });
});
