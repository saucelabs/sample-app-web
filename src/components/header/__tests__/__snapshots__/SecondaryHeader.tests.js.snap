// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SecondaryHeader should render all props 1`] = `
<div
  className="header_secondary_container"
>
  <LeftComponent
    leftComponent={
      <bound Template
        type="checkbox"
      />
    }
  />
  <Title
    title="sample title"
  />
  <HeaderBot />
  <RightComponent
    rightComponent={
      <bound Template
        type="radio"
      />
    }
  />
</div>
`;

exports[`SecondaryHeader should render without any props 1`] = `
<div
  className="header_secondary_container"
/>
`;
